python/strings_udf/strings_udf/_typing.py

```diff
@@ -112,7 +114,9 @@ def prepare_args(self, ty, val, **kwargs):
         if isinstance(ty, types.CPointer) and isinstance(
             ty.dtype, (StringView, UDFString)
         ):
-            return types.uint64, val.ptr
+            return types.uint64, val.ptr if isinstance(
+                val, rmm._lib.device_buffer.DeviceBuffer
+            ) else val.get_ptr(mode="read")
```

Comment:
This will expose any buffer passed in permanently unless someone is holding a spill lock, but I don't know who should be doing so? Presumably it is the caller of the strings_udf kernel. Where is that @brandon-b-miller ?

Reply:
two places:
https://github.com/rapidsai/cudf/blob/branch-23.02/python/strings_udf/strings_udf/tests/test_string_udfs.py#L71
https://github.com/rapidsai/cudf/blob/branch-23.02/python/cudf/cudf/core/indexed_frame.py#L2131